packages/react-scripts/config/webpack.config.js

```diff
@@ -319,11 +319,12 @@ module.exports = function(webpackEnv) {
                 formatter: require.resolve('react-dev-utils/eslintFormatter'),
                 eslintPath: require.resolve('eslint'),
                 // @remove-on-eject-begin
-                baseConfig: {
+                baseConfig: !process.env.RS_ESLINTRC && {
                   extends: [require.resolve('eslint-config-react-app')],
                 },
+                useEslintrc: !!process.env.RS_ESLINTRC,
```

Comment:
`RS_ESLINTRC` is the main point of this PR, when this env var is defined, baseConfig is not used, and a local .eslintrc is used